exchanges/poloniex/poloniex.go

```diff
@@ -16,920 +16,1162 @@ import (
+func (e *Exchange) CancelSmartOrderByID(ctx context.Context, id, clientSuppliedID string) (*CancelSmartOrderResponse, error) {
+	path, err := orderPath(id, "/smartorders/", clientSuppliedID, "/smartorders/cid:")
 	if err != nil {
-		return false, err
+		return nil, err
 	}
-
-	if result.Success == 0 {
-		return false, errors.New(result.Error)
+	var cancelSmartOrderResponse *CancelSmartOrderResponse
+	resp := &V3ResponseWrapper{
+		Data: &cancelSmartOrderResponse,
 	}
-
-	return true, nil
+	if err := e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, sCancelSmartOrderByIDEPL, http.MethodDelete, path, nil, nil, &resp); err != nil {
+		return nil, err
+	}
+	if resp.Code != 0 && resp.Code != 200 {
+		return nil, fmt.Errorf("%w code: %d message: %s", request.ErrAuthRequestFailed, resp.Code, resp.Msg)
+	}
+	if cancelSmartOrderResponse == nil {
+		return nil, common.ErrNoResponse
+	} else if cancelSmartOrderResponse.Code != 0 && cancelSmartOrderResponse.Code != 200 {
+		ordID := id
+		if ordID == "" {
+			ordID = clientSuppliedID
+		}
+		return cancelSmartOrderResponse, fmt.Errorf("%w: order ID: %s code: %d message: %s", order.ErrCancelFailed, ordID, cancelSmartOrderResponse.Code, cancelSmartOrderResponse.Message)
+	}
+	return cancelSmartOrderResponse, nil
 }
 
-// GetOpenLoanOffers returns all open loan offers
-func (e *Exchange) GetOpenLoanOffers(ctx context.Context) (map[string][]LoanOffer, error) {
-	type Response struct {
-		Data map[string][]LoanOffer
+// CancelMultipleSmartOrders performs a batch cancel one or many smart orders in an account by IDs.
+func (e *Exchange) CancelMultipleSmartOrders(ctx context.Context, args *CancelOrdersRequest) ([]*CancelOrderResponse, error) {
+	if args == nil {
+		return nil, common.ErrNilPointer
 	}
-	result := Response{}
-
-	err := e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, http.MethodPost, poloniexOpenLoanOffers, url.Values{}, &result.Data)
-	if err != nil {
-		return nil, err
+	if len(args.ClientOrderIDs) == 0 && len(args.OrderIDs) == 0 {
+		return nil, order.ErrOrderIDNotSet
 	}
-
-	if result.Data == nil {
-		return nil, errors.New("there are no open loan offers")
+	var cancelResponses []*CancelOrderResponse
+	resp := &V3ResponseWrapper{
+		Data: &cancelResponses,
 	}
-
-	return result.Data, nil
+	return cancelResponses, e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, sCancelSmartOrdersByIDEPL, http.MethodDelete, "/smartorders/cancelByIds", nil, args, &resp)
 }
 
-// GetActiveLoans returns active loans
-func (e *Exchange) GetActiveLoans(ctx context.Context) (ActiveLoans, error) {
-	result := ActiveLoans{}
-	return result, e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, http.MethodPost, poloniexActiveLoans, url.Values{}, &result)
+// CancelSmartOrders cancels all smart orders in an account.
+func (e *Exchange) CancelSmartOrders(ctx context.Context, symbols, accountTypes, orderTypes []string) ([]*CancelOrderResponse, error) {
```

Comment:
🚧 Change request:
Whilst providing a patch for orderTypes, I realised this should also be accepting []accountType and probably []currency.Pair instead of strings.